### PR TITLE
Add docs for GenerateContentResponse

### PR DESCRIPTION
**Pull request by morganchen12**

### CODE REVIEW

File: Sources/GoogleAI/GenerateContentResponse.swift [hunk 1]
```diff
@@ -135,16 +152,28 @@ extension CandidateResponse: Decodable {
...
 public struct CitationMetadata: Decodable {
+
+  /// A list of individual cited sources and the parts of the content to which they apply.
   public let citationSources: [Citation]
 }
 
+/// A struct describing a source attribution.
 public struct Citation: Decodable {
+
+  /// The beginning of a sequence in a model response, inclusive, that derives from a cited source.
```
Comment (paulb777) [inline Sources/GoogleAI/GenerateContentResponse.swift:163]: My preference would be:

The inclusive beginning of a sequence in a model response that derives from a cited source.

File: Sources/GoogleAI/GenerateContentResponse.swift [hunk 2]
```diff
@@ -135,16 +152,28 @@ extension CandidateResponse: Decodable {
...
   public let startIndex: Int
+
+  /// The end of a sequence in a model response, exclusive, that derives from a cited source.
```
Comment (paulb777) [inline Sources/GoogleAI/GenerateContentResponse.swift:166]: And:

The exclusive end of a sequence in a model response that derives from a cited source.

File: Sources/GoogleAI/GenerateContentResponse.swift [hunk 3]
```diff
@@ -160,7 +189,11 @@ public enum FinishReason: String {
   /// NOTE: When streaming, the Candidate.content will be empty if content filters blocked the
   /// output.
   case safety = "SAFETY"
+
+  /// The token generation was stopped as the response was flagged for unauthorized citations.
```
Comment (paulb777) [inline Sources/GoogleAI/GenerateContentResponse.swift:193]: s/as/because/

**Review by paulb777 — APPROVED** [review]
Some optional rewording suggestions and please get the PR green

Thanks!

File: Sources/GoogleAI/GenerateContentResponse.swift (hunk 1 above)
Comment (morganchen12) [reply Sources/GoogleAI/GenerateContentResponse.swift]: Done.

File: Sources/GoogleAI/GenerateContentResponse.swift (hunk 2 above)
Comment (morganchen12) [reply Sources/GoogleAI/GenerateContentResponse.swift]: Done.

File: Sources/GoogleAI/GenerateContentResponse.swift (hunk 3 above)
Comment (morganchen12) [reply Sources/GoogleAI/GenerateContentResponse.swift]: Done.

**Pull request merged**